### ISSUE TITLE
[Issue]: IOS datasets Salinity variables to CTD specific BODC 

### ISSUE THREAD

**Issue by JessyBarrette** [cioos-siooc/cioos-pacific-erddap#62]
### Level

High

### Issue Description

A lot of the different CTD datasets from IOS rigth now are refering the Salinity variables with are identified within the IOS files as:

- salinity 
- salinity: pre-1978
- salinity:bottle
- salinity:t0:c0
- salinity:t1:c1

As `SDN:P01::PSALST01` : Practical salinity of the water body by CTD and computation using UNESCO 1983 algorithm
while often the first three variable names are referering salinity measurement made from bottles.

This is applies to the following datasets:
- #2 
- #16
- #3 

### Issue Solution

It would be more appripropriate to attach those different values to a more appropriate BODC term like the followings:
practical salinity
- [SDN:P01::PSALBSTX](http://vocab.nerc.ac.uk/collection/P01/current/PSALBSTX): Practical salinity of the water body by bench salinometer and computation using UNESCO 1983 algorithm
- [SDN:P01::SSALBSTX](http://vocab.nerc.ac.uk/collection/P01/current/SSALBSTX): Salinity of the water body by bench salinometer

and/or a more broader terms:
- [SDN:P01::PSALZZXX](http://vocab.nerc.ac.uk/collection/P01/current/PSALZZXX): Practical salinity of the water body by computation using UNESCO 1983 algorithm
- [SDN:P01::ODSDM021](http://vocab.nerc.ac.uk/collection/P01/current/ODSDM021): Salinity of the water body

We can also improve the mapping of the different ctd specific salinity variables by using a combination of variable names and instrument.

**Comment by guanlu129** [comment]
salinity:t0:c0 is from CTD measurement. Should be appropriated to use SDN:P01::PSALST01?
salinity:t1:c1 is also from CTD measurement. Should be appropriate to use SDN:P01::PSALST01?

**Comment by guanlu129** [comment]
no variable is shown as salinity in OSD data archive.

**Comment by guanlu129** [comment]
UBC and MEDS files use salinity: pre-1978 in unit of ppt. 
There's limited information in the files about how the salinity was measured. 
Should we use the broader term: [SDN:P01::ODSDM021]: Salinity of the water body for UBC and MEDS files?

**Comment by guanlu129** [comment]
salinity:bottle indicates salinity measurements from bottle samples. 
There are three salinity variables from bottle samples:
salinity:bottle ()
salinity:bottle (PSS-78)
salinity:bottle (ppt)

So should be appropriate to use [SDN:P01::SSALBSTX]: Salinity of the water body by bench salinometer for salinity:bottle ()?

**Comment by JessyBarrette** [comment]
As of now all salinity data in units of PPT gets mapped to a not unit specific BODC like
- SDN:P01::SSALST01: Salinity of the water body by CTD
- SDN:P01::SSALBSTX: Salinity of the water body by bench salinometer

None is mapped to OSDDM021 yet but we could add them.

All PPT  and pre-1978 units are converted to practical salinity by using the TEOS-10 equation. `gsw.SP_from_SK`

**Comment by JessyBarrette** [comment]
Here's the full list of salinity variables detected 

variable | units
-- | --
salinity | PSS\-78\|PPS78
salinity | ppt\|PPT
salinity | ppt\|PPT
salinity: practical | PSS-78\|PSU
salinity: pre-1978 | PPT
salinity: pre-1978 | PPT
salinity:bottle | PPT\|ppt
salinity:bottle | PPT
salinity:bottle | PSS-78
salinity:bottle |  
salinity:ctd | PSS-78
salinity:cur | PSS-78
salinity:cur | ppt
salinity:t0:c0 | none\|PSS-78
salinity:t1:c1 | PSS-78
salinity:upcast | PSS-78

**Comment by guanlu129** [comment]
> 

Two questions to confirm:
1) could we map multiple BODC standard parameters for each variable in ios data file?
2)  "All PPT and pre-1978 units are converted to practical salinity by using the TEOS-10 equation. gsw.SP_from_SK" - does this happen in file conversion?

**Comment by JessyBarrette** [comment]
> 1. could we map multiple BODC standard parameters for each variable in ios data file?
yes to both of them. Right now we're using this to generate multiple BODCs for temperature data as an example and planning to do the same for the salinity.

> 2. "All PPT and pre-1978 units are converted to practical salinity by using the TEOS-10 equation. gsw.SP_from_SK" - does this happen in file conversion?

Yes, it would happend there.

Generally the new netcdf files would contains more variables than the original ios file to handle the different BODC terms.
